src/cursor/cur_log.c

```diff
@@ -238,8 +238,11 @@ __curlog_next(WT_CURSOR *cursor)
 	 */
 	if (cl->stepp == NULL || cl->stepp >= cl->stepp_end || !*cl->stepp) {
 		cl->txnid = 0;
-		WT_ERR(__wt_log_scan(session, cl->next_lsn, WT_LOGSCAN_ONE,
-		    __curlog_logrec, cl));
+		ret = __wt_log_scan(session, cl->next_lsn, WT_LOGSCAN_ONE,
+		    __curlog_logrec, cl);
+		if (ret == ENOENT)
+			ret = WT_NOTFOUND;
```

Comment:
I think this one should set ret = 0.  The `next_lsn` is simply set to the end of the current record which could be the last one in a log file.  We want to continue and move onward from that.  The `search` one below is correct.